.github/workflows/kube-linter.yaml

```diff
@@ -0,0 +1,45 @@
+name: Check Kubernetes YAMLs with kube-linter
+
+on:
+  # Note that both `push` and `pull_request` triggers should be present for GitHub to consistently present kube-linter
+  # SARIF reports.
+  pull_request:
+    branches: [ main ]
+jobs:
+  scan:
+    permissions: write-all
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      # This prepares directory where github/codeql-action/upload-sarif@v1 looks up report files by default.
+      - name: Create ../results directory for SARIF report files
+        shell: bash
+        run: mkdir -p ../results
+
+      - name: Scan yaml files with kube-linter
```

Comment:
Some of the resources we are deploying are located on remote repositories, or generated after applying a pathc, so we only get them when running `kustomize build`. I think that we should first build all the manifests, dump them to a directory and then run the check.

Reply:
updated [in this commit](https://github.com/redhat-appstudio/infra-deployments/pull/2197/commits/36cbf1e1b367b84bb94c5ad9e1da94a2e0a82cc6), please check (now we have [470 warnings](https://github.com/redhat-appstudio/infra-deployments/security/code-scanning?query=pr%3A2197+tool%3Akube-linter+is%3Aopen), yay :) 

however there were numerous errors hit [during `kustomize build` step](https://github.com/redhat-appstudio/infra-deployments/actions/runs/5750076950/job/15586110735?pr=2197#step:5:7) (search for `^ ERROR`) - should we address them in this PR or create a follow-up ticket?

Reply:
I suggest to address them in a follow up pr.

Reply:
https://issues.redhat.com/browse/RHTAPSRE-109